test/LidoStakedEthRewardsIntegration.t.sol

```diff
@@ -0,0 +1,116 @@
+
+    function setUp() public {
+        vm.createSelectFork(getChain("mainnet").rpcUrl, 18_707_715);  // Dec 3rd, 2023
+
+        transferStrategy = new PullRewardsTransferStrategy(
+            address(incentives),
+            admin,
+            operator
+        );
+
+        deal(WSTETH, operator, REWARD_AMOUNT);
+
+        vm.prank(operator);
+        IERC20(WSTETH).approve(address(transferStrategy), REWARD_AMOUNT);
+
+        vm.prank(admin);
+        emissionManager.setEmissionAdmin(WSTETH, operator);
+    }
+
+    function _setupDistribution() internal {
+        RewardsDataTypes.RewardsConfigInput[] memory configs = new RewardsDataTypes.RewardsConfigInput[](1);
+
+        configs[0] = RewardsDataTypes.RewardsConfigInput({
+            emissionPerSecond: uint88(REWARD_AMOUNT / DURATION),
+            totalSupply:       0,  // Set by the rewards controller
+            distributionEnd:   uint32(block.timestamp + DURATION),
+            asset:             _getAToken(WETH),  // Rewards on WETH supplies
+            reward:            WSTETH,
+            transferStrategy:  transferStrategy,
+            rewardOracle:      IEACAggregatorProxy(aaveOracle.getSourceOfAsset(WSTETH))
+        });
+
+        vm.prank(operator);
+        emissionManager.configureAssets(configs);
+    }
+
+    function test_setup_distribution() public {
+        _setupDistribution();
+    }
+
+    function test_user_claim() public {
+        address claimAddress = makeAddr("claimAddress");
+        address[] memory assets = new address[](1);
+        assets[0] = _getAToken(WETH);
+
+        vm.prank(whale);
+        incentives.claimAllRewards(assets, claimAddress);
+        assertEq(IERC20(WSTETH).balanceOf(claimAddress), 0);
+
+        _setupDistribution();
+
+        vm.prank(whale);
+        incentives.claimAllRewards(assets, claimAddress);
+        assertEq(IERC20(WSTETH).balanceOf(claimAddress), 0);
+
+        uint256 skipAmount = DURATION / 2;  // 50% of rewards distrubuted
+        skip(skipAmount);
+
+        // 7-siblings wallet should get about half of the rewards at this time
+        // 79k ETH deposit out of 157k total supplied
+        vm.prank(whale);
+        incentives.claimAllRewards(assets, claimAddress);
+        uint256 whaleReward1 = 5.028734135612479150 ether;
+        assertEq(IERC20(WSTETH).balanceOf(claimAddress), whaleReward1);
+        assertEq(IERC20(WSTETH).balanceOf(operator),     REWARD_AMOUNT - whaleReward1);
+
+        skip(DURATION - skipAmount);
+
+        // 7-siblings wallet should get about 50% of the total rewards ~10 wstETH
+        vm.prank(whale);
+        incentives.claimAllRewards(assets, claimAddress);
+        uint256 whaleReward2 = 10.057468271224958300 ether;
+        assertEq(whaleReward2, whaleReward1 * 2);
+        assertEq(IERC20(WSTETH).balanceOf(claimAddress), whaleReward2);
+        assertEq(IERC20(WSTETH).balanceOf(operator),     REWARD_AMOUNT - whaleReward2);
```

Comment:
Lets add another code block to warp more and demonstrate that the rewards stop accruing after a certain amount of time.

Lets also add a ticket to make a test with multiple users to make sure that the distribution is equal for everyone